# Ok Cupid Dataset EDA

This dataset was picked on the name alone to do an initial clean and EDA.
The dataset is completly unfamiliar to me - I want to practice familiarising myself with the set and seeing what I can learn about it, to see what further investigation can stem from it. 

In [26]:
# initial imports and check file is readable
import pandas as pd
okcupid = pd.read_csv('OkCupid_kaggle_dataset.zip')


In [27]:
pd.set_option('display.max_columns', None) # use this to display ALL columns and not get a ... midway
print(okcupid.head((10)))

   age     status sex orientation       body_type               diet  \
0   22     single   m    straight  a little extra  strictly anything   
1   35     single   m    straight         average       mostly other   
2   38  available   m    straight            thin           anything   
3   23     single   m    straight            thin         vegetarian   
4   29     single   m    straight        athletic                NaN   
5   29     single   m    straight         average    mostly anything   
6   32     single   f    straight             fit  strictly anything   
7   31     single   f    straight         average    mostly anything   
8   24     single   f    straight             NaN  strictly anything   
9   37     single   m    straight        athletic    mostly anything   

       drinks      drugs                          education  \
0    socially      never      working on college/university   
1       often  sometimes              working on space camp   
2    socially     

In [28]:
okcupid.shape

(59946, 31)

This set has 59946 rows and 31 columns

## Duplicates check

In [29]:
duplicates = okcupid.duplicated()
print(duplicates.value_counts())

False    59946
dtype: int64


The number of false duplicates equals the number of rows - i.e. there are no dupicated rows

## Tidying/Wrangling

In [30]:
print(okcupid.columns)

Index(['age', 'status', 'sex', 'orientation', 'body_type', 'diet', 'drinks',
       'drugs', 'education', 'ethnicity', 'height', 'income', 'job',
       'last_online', 'location', 'offspring', 'pets', 'religion', 'sign',
       'smokes', 'speaks', 'essay0', 'essay1', 'essay2', 'essay3', 'essay4',
       'essay5', 'essay6', 'essay7', 'essay8', 'essay9'],
      dtype='object')


In [64]:
okcupid_variables = pd.DataFrame(okcupid.info())

# okcupid_variables = pd.DataFrame(okcupid.info()) #converted to a DataFrame to send to an Excel file


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 59946 entries, 0 to 59945
Data columns (total 31 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   age          59946 non-null  int64  
 1   status       59946 non-null  object 
 2   sex          59946 non-null  object 
 3   orientation  59946 non-null  object 
 4   body_type    54650 non-null  object 
 5   diet         35551 non-null  object 
 6   drinks       56961 non-null  object 
 7   drugs        45866 non-null  object 
 8   education    53318 non-null  object 
 9   ethnicity    54266 non-null  object 
 10  height       59943 non-null  float64
 11  income       59946 non-null  int64  
 12  job          51748 non-null  object 
 13  last_online  59946 non-null  object 
 14  location     59946 non-null  object 
 15  offspring    24385 non-null  object 
 16  pets         40025 non-null  object 
 17  religion     39720 non-null  object 
 18  sign         48890 non-null  object 
 19  smok

As there are alot of variables to consider types and further actions for, I'd like to track these in an Excel sheet
To do this, I plan on exporting okcupid.info() to an Excel sheet directly

In [57]:
#from xlsxwriter import Workbook
#okcupid_variables.to_excel('okcupid_variables.xlsx', engine='xlsxwriter')
#okcupid_variables.to_csv('okcupid_variables.csv', sep=',', header=True) # works but loses all the data

In [63]:
#print(okcupid_variables.head()) # shows okcupid_variables is actually empty, so I didnt lose data in the CSV there just isnt any

Empty DataFrame
Columns: []
Index: []


This is something I want to keep in mind but will shelve for now as it'll take me more time to learn than save me at the moment.

Next I'll inspect each variable

0 'age' appears complete with Dtype int64 - ready to investiagte numerically
1 'status' - complete, object - check the unique types with .value_counts()
2 & 3 'sex' & 'orientation' - complete, object - check the unique types with .value_counts()

4 'body_type' - 54650 entries, therefore incomplete - check the unique types with .value_counts()
5 'diet' - 35551 - .value_counts()


In [65]:
okcupid.status.value_counts()

single            55697
seeing someone     2064
available          1865
married             310
unknown              10
Name: status, dtype: int64

'Status' is Relationship Status.
Categories are:
* Single
* Seeing somone
* Available
* Married
* Unknown - this one needs further investigation: unknown as in profile hasn't added their status? (from code snippet below I'd say yes)

In [73]:
#status_unknowns = okcupid[okcupid.status=='unknown']
#print(status_unknowns)
#print(len(status_unknowns))


In [74]:
okcupid.sex.value_counts()

m    35829
f    24117
Name: sex, dtype: int64

Sex is only in two vales: 'm' or 'f'

In [76]:
okcupid.orientation.value_counts()

straight    51606
gay          5573
bisexual     2767
Name: orientation, dtype: int64

Orientation is 'straight', 'gay' and 'bisexual'
Seen here large majority is straight, followed by gay then bisexual

In [77]:
okcupid.body_type.value_counts()

average           14652
fit               12711
athletic          11819
thin               4711
curvy              3924
a little extra     2629
skinny             1777
full figured       1009
overweight          444
jacked              421
used up             355
rather not say      198
Name: body_type, dtype: int64

average           14652, 
fit               12711, 
athletic          11819, 
thin               4711, 
curvy              3924, 
a little extra     2629, 
skinny             1777, 
full figured       1009, 
overweight          444, 
jacked              421, 
used up             355, 
rather not say      198

In [61]:
#okcupid.dtypes # done in .info()


In [58]:
#okcupid.describe()